src/plugins/discover/public/application/angular/doc_table/doc_table.ts

```diff
@@ -1,98 +0,0 @@
-/*
- * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
- * or more contributor license agreements. Licensed under the Elastic License
- * 2.0 and the Server Side Public License, v 1; you may not use this file except
- * in compliance with, at your election, the Elastic License 2.0 or the Server
- * Side Public License, v 1.
- */
-
-import html from './doc_table.html';
-import { dispatchRenderComplete } from '../../../../../kibana_utils/public';
-import { SAMPLE_SIZE_SETTING } from '../../../../common';
-// @ts-expect-error
-import { getLimitedSearchResultsMessage } from './doc_table_strings';
-import { getServices } from '../../../kibana_services';
-import './index.scss';
-
-export interface LazyScope extends ng.IScope {
-  // eslint-disable-next-line @typescript-eslint/no-explicit-any
-  [key: string]: any;
-}
-
-// eslint-disable-next-line @typescript-eslint/no-explicit-any
-export function createDocTableDirective(pagerFactory: any, $filter: any) {
-  return {
-    restrict: 'E',
-    template: html,
-    scope: {
-      sorting: '=',
-      columns: '=',
-      hits: '=',
-      totalHitCount: '=',
-      indexPattern: '=',
-      isLoading: '=?',
-      infiniteScroll: '=?',
-      filter: '=?',
-      minimumVisibleRows: '=?',
-      onAddColumn: '=?',
-      onChangeSortOrder: '=?',
-      onMoveColumn: '=?',
-      onRemoveColumn: '=?',
-      inspectorAdapters: '=?',
-      useNewFieldsApi: '<',
-    },
-    link: ($scope: LazyScope, $el: JQuery) => {
-      $scope.persist = {
-        sorting: $scope.sorting,
-        columns: $scope.columns,
-      };
-
-      const limitTo = $filter('limitTo');
-      const calculateItemsOnPage = () => {
-        $scope.pager.setTotalItems($scope.hits.length);
-        $scope.pageOfItems = limitTo($scope.hits, $scope.pager.pageSize, $scope.pager.startIndex);
-      };
-
-      $scope.limitedResultsWarning = getLimitedSearchResultsMessage(
-        getServices().uiSettings.get(SAMPLE_SIZE_SETTING, 500)
-      );
-
-      $scope.addRows = function () {
-        $scope.limit += 50;
-      };
-
-      $scope.$watch('minimumVisibleRows', (minimumVisibleRows: number) => {
-        $scope.limit = Math.max(minimumVisibleRows || 50, $scope.limit || 50);
-      });
-
-      $scope.$watch('hits', (hits: unknown[]) => {
-        if (!hits) return;
-
-        // Reset infinite scroll limit
-        $scope.limit = $scope.minimumVisibleRows || 50;
-
-        if (hits.length === 0) {
-          dispatchRenderComplete($el[0]);
```

Comment:
Do we actually need `renderComplete` directive? or we can remove it

Reply:
if can be removed, `data-render-complete` is needed for Dashboard tests and can be set like this: 

`data-render-complete={isLoading ? 'false' : 'true'}`

https://github.com/elastic/kibana/blob/80b109f95c344c8e7f8d484242f4be2feb32ba3c/src/plugins/discover/public/application/components/discover_grid/discover_grid.tsx#L372

Reply:
Nicer would be `data-render-complete={!isLoading}`

Reply:
Nicer would be `data-render-complete={!isLoading}`, since it will be transformed into a string

Reply:
totally agree :)